src/Compiler/SyntaxTree/SyntaxTreeOps.fs

```diff
@@ -463,14 +471,8 @@ let mkSynDot mDot m l (SynIdent (r, rTrivia)) =
         SynExpr.DotGet(e, dm, SynLongIdent(lid @ [ r ], dots @ [ mDot ], trivia @ [ rTrivia ]), m)
     | expr -> SynExpr.DotGet(expr, mDot, SynLongIdent([ r ], [], [ rTrivia ]), m)
 
-let mkSynDotMissing mDot m l =
-    match l with
-    | SynExpr.LongIdent (isOpt, SynLongIdent (lid, dots, trivia), None, _) ->
-        // REVIEW: MEMORY PERFORMANCE: This list operation is memory intensive (we create a lot of these list nodes)
-        SynExpr.LongIdent(isOpt, SynLongIdent(lid, dots @ [ mDot ], trivia), None, m)
-    | SynExpr.Ident id -> SynExpr.LongIdent(false, SynLongIdent([ id ], [ mDot ], [ None ]), None, m)
-    | SynExpr.DotGet (e, dm, SynLongIdent (lid, dots, trivia), _) -> SynExpr.DotGet(e, dm, SynLongIdent(lid, dots @ [ mDot ], trivia), m) // REVIEW: MEMORY PERFORMANCE: This is memory intensive (we create a lot of these list nodes)
-    | expr -> SynExpr.DiscardAfterMissingQualificationAfterDot(expr, m)
+let mkSynDotMissing (mDot: range) (m: range) (expr: SynExpr) =
+    SynExpr.DiscardAfterMissingQualificationAfterDot(expr, mDot, unionRanges mDot m)
```

Comment:
I guess I could just inline this function into the grammar rules now?

Reply:
Or maybe not. There're similar one line functions. I think these should be changed together in a separate PR if we decide to remove them.